internal/app/api/handlers/balance.go

```diff
@@ -0,0 +1,77 @@
+package handlers
+
+import (
+	"encoding/json"
+	"github.com/northmule/gophermart/internal/app/api/rctx"
+	"github.com/northmule/gophermart/internal/app/repository"
+	"github.com/northmule/gophermart/internal/app/repository/models"
+	"github.com/northmule/gophermart/internal/app/services/logger"
+	"net/http"
+)
+
+type BalanceHandler struct {
+	manager repository.Repository
+}
+
+func NewBalanceHandler(manager repository.Repository) *BalanceHandler {
+	instance := &BalanceHandler{
+		manager: manager,
+	}
+	return instance
+}
+
+type responseBalance struct {
+	Current   float64 `json:"current"`
+	Withdrawn float64 `json:"withdrawn"`
+}
+
+func (bh *BalanceHandler) Balance(res http.ResponseWriter, req *http.Request) {
+	user := req.Context().Value(rctx.UserCtxKey).(models.User)
+
+	balance, err := bh.manager.Balance().FindOneByUserUUID(user.UUID)
+	if err != nil {
+		logger.LogSugar.Errorf(err.Error())
+		res.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	withdrawn, err := bh.manager.Withdrawn().FindSumWithdrawnByUserUUID(user.UUID)
+	if err != nil {
+		logger.LogSugar.Errorf(err.Error())
+		res.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	response := responseBalance{
+		Current:   0,
+		Withdrawn: withdrawn,
+	}
+	if balance != nil {
+		response.Current = balance.Value
+	}
+
+	responseBalanceValue, err := json.Marshal(response)
+	if err != nil {
+		http.Error(res, "Ошибка подготовки ответа", http.StatusInternalServerError)
+		logger.LogSugar.Error(err.Error())
```

Comment:
Стоит добавить контекста к сообщению об ошибке. zap.Error(err) - добавить в лог поле `error`.
```suggestion
		logger.LogSugar.Error("маршаллинг ответа", zap.Error(err))
```